homeassistant/helpers/script.py

```diff
@@ -130,39 +142,43 @@ def __init__(
         sequence: Sequence[Dict[str, Any]],
         name: Optional[str] = None,
         change_listener: Optional[Callable[..., Any]] = None,
+        mode: Optional[str] = None,
     ) -> None:
         """Initialize the script."""
         self.hass = hass
         self.sequence = sequence
         template.attach(hass, self.sequence)
         self.name = name
         self._change_listener = change_listener
-        self._cur = -1
-        self._exception_step: Optional[int] = None
-        self.last_action = None
+        self._runs: List[Script._ScriptRun] = []
         self.last_triggered: Optional[datetime] = None
         self.can_cancel = any(
             CONF_DELAY in action or CONF_WAIT_TEMPLATE in action
             for action in self.sequence
         )
-        self._async_listener: List[CALLBACK_TYPE] = []
         self._config_cache: Dict[Set[Tuple], Callable[..., bool]] = {}
-        self._actions = {
-            ACTION_DELAY: self._async_delay,
-            ACTION_WAIT_TEMPLATE: self._async_wait_template,
-            ACTION_CHECK_CONDITION: self._async_check_condition,
-            ACTION_FIRE_EVENT: self._async_fire_event,
-            ACTION_CALL_SERVICE: self._async_call_service,
-            ACTION_DEVICE_AUTOMATION: self._async_device_automation,
-            ACTION_ACTIVATE_SCENE: self._async_activate_scene,
-        }
+        self._mode = (
+            mode
+            if mode
```

Comment:
Please write this as a normal if…elif statement. We should only use inline if-statements if they fit on one line.

Reply:
Will do.